book/src/design/circuit/gadgets/decomposition.md

```diff
@@ -22,25 +22,51 @@ $$
 Strict mode constrains the running sum output $z_{W}$ to be zero, thus range-constraining the field element to be within $W \cdot K$ bits.
 
 In strict mode, we are also assured that $z_{W-1} = k_{W-1}$ gives us the last window in the decomposition.
-
 ## Lookup decomposition
 This gadget makes use of a $K$-bit lookup table to decompose a field element $\alpha$ into $K$-bit words. Each $K$-bit word $k_i = z_i - 2^K \cdot z_{i+1}$ is range-constrained by a lookup in the $K$-bit table.
 
+The region layout for the lookup decomposition uses a single advice column $z$, and two selectors $q_{lookup}$ and $q_{running}.$
+$$
+\begin{array}{|c|c|c|}
+\hline
+    z    & q_{lookup} & q_{running} \\\hline
+  z_0    &     1      &       1     \\\hline
+  z_1    &     1      &       1     \\\hline
+\vdots   &   \vdots   &     \vdots  \\\hline
+z_{n-1}  &     1      &       1     \\\hline
+z_n      &     0      &       0     \\\hline
+\end{array}
+$$
 ### Short range check
 Using two $K$-bit lookups, we can range-constrain a field element $\alpha$ to be $n$ bits, where $n \leq K.$ To do this:
 
 1. Constrain $0 \leq \alpha < 2^K$ to be within $K$ bits using a $K$-bit lookup.
 2. Constrain $0 \leq \alpha \cdot 2^{K - n} < 2^K$ to be within $K$ bits using a $K$-bit lookup.
 
-## Short range decomposition
-For a short range (for instance, $2^K \leq 8$), we can range-constrain each word using a polynomial constraint instead of a lookup:
+The short variant of the lookup decomposition introduces a $q_{bitshift}$ selector. The same advice column $z$ has here been renamed to $\textsf{word}$ for clarity:
+$$
+\begin{array}{|c|c|c|c|}
+\hline
+\textsf{word} & q_{lookup} & q_{running} & q_{bitshift} \\\hline
+\alpha        &     1      &      0      &       0      \\\hline
+\alpha'       &     1      &      0      &       1      \\\hline
+2^{K-n}       &     0      &      0      &       0      \\\hline
```

Comment:
If we used two advice columns here, then we could put 2<sup>K-n</sup> in the first row of the second column, and then we'd be able to refer to all three cells using only *cur* and *next* queries. I don't want to delay this PR getting in, so this can be a later optimization if and when we try to optimize other gates to only use *cur* and *next*.